src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -129,6 +140,25 @@ public static String jenkinsJobName(BuildConfig bc) {
     return jenkinsJobName(namespace, name);
   }
 
+  // Since the token we need can change without updating the global config, we need to get it every time we do a call
+  public static void setOpenShiftOauthToken() {
+    GlobalPluginConfiguration config = GlobalConfiguration.all().get(GlobalPluginConfiguration.class);
+    if(config == null){
+       return;
+    }
+    String credentialsId = config.getCredentialsId();
+    List<OpenShiftTokenCredentials> tokens = CredentialsProvider.lookupCredentials(OpenShiftTokenCredentials.class, (ItemGroup)null, ACL.SYSTEM , (DomainRequirement)null);
+    if( tokens != null && tokens.size() > 0 ) {
+      for (OpenShiftTokenCredentials token : tokens) {
+        if (token.getId().equals(credentialsId)) {
+          getOpenShiftClient().getConfiguration().setOauthToken(token.getToken());
```

Comment:
It is recursive, and I'm confused why this didn't cause an overflow when I was testing. It's unnecessary anyways, because I can just grab the actual client w/o a getter